File: src/trabajo/src/informacion_personal_nodo.cpp

#include "ros/ros.h"
#include "trabajo/inf_personal_usuario.h"
//#include <vector>
#include "std_msgs/String.h"

using namespace std;

trabajo::inf_personal_usuario datos;

std_msgs::String cadena[2];
std_msgs::String cad;


int main(int argc, char **argv)
{
    ros::init(argc, argv, "inf_personal_usuario");
    ros::NodeHandle n;
    ros::Publisher pub = n.advertise<trabajo::inf_personal_usuario>("inf_pers_topic", 1000);
    ROS_INFO("nodo inf_personal_usuario iniciado!");

    string nombre;
    int edad;
    //string idiomas;
    int aux=0;
    //string auxIdioma;    

    while (ros::ok())
    {
        ROS_INFO("Introducir informacion personal usuario");
        cout<<"Introducir nombre: ";
        getline(cin,nombre);
        //cad.data = nombre;

        datos.nombre = nombre;
        cout<<"Introducir edad: ";
        cin >> edad;
        datos.edad = edad;
        cout<<"Cuantos idiomas habla?\n";
        cin >> aux;
        //vector<string> idiomas[aux];
        int i = 0;
        string auxIdioma;
        while(i<aux)
        {
            cout<<"Introducir idioma "+(i+1);
            getline(cin,auxIdioma);
            cad.data = auxIdioma;
            cadena[0] = cad;
            i+= 1;
        }
        
        datos.idiomas = &cadena;
        pub.publish(datos);
    }
    return 0;
}

File: src/trabajo/scripts/informacion_personal_nodo.py
#!/usr/bin/env python
# Se importa modulos de ROS
import rospy
# Se importa modulos de dependencias de mensajes del directorio msg
from trabajo.msg import inf_personal_usuario
from espeak import espeak

# Varialbe para almacenar los idiomas de usuario


def Publicador():
    # Se inicia el nodo publicador donde envia mensajes 
    pub = rospy.Publisher('inf_pers_topic', inf_personal_usuario, queue_size=10)
    rospy.init_node('inf_personal_usuario', anonymous=True)
    rate = rospy.Rate(1)

    while not rospy.is_shutdown():
        # Obtenermos la informacion de usuario
        idiomas_aux = []
        datos_usuario = inf_personal_usuario()
        rospy.loginfo("Introducir informacion de usuario")
        n = raw_input("Ingrese su nombre: ")
        e = int(raw_input("Ingrese su edad: "))
        n_idiomas = int(raw_input("Cuantos idiomas habla?: "))
        # Un bucle para pedir segun la cantidad de numero de idiomas del usuario
        for i in range(n_idiomas):
            idioma = raw_input("Ingrese su idioma "+str(i+1)+": ")
            idiomas_aux.append(idioma)
        # Guardamos las variables en el archivo msg
        datos_usuario.nombre = n
        datos_usuario.edad = e
        datos_usuario.idiomas = idiomas_aux
        # Notificamos y se publican los datos
        rospy.loginfo("Datos usuarios publicados!")
        pub.publish(datos_usuario)
        rate.sleep()

if __name__ == '__main__':
    try:
        Publicador()
    except rospy.ROSInterruptException:
        pass